lib/pool.js

```diff
@@ -0,0 +1,110 @@
+
+const debug                = require('debug')('kuss:pool')
+const R                    = require('ramda')
+const Bluebird             = require('bluebird')
+const GenericPool          = require('generic-pool')
+const { FibonacciBackoff } = require('simple-backoff')
+
+const CONFIG_PROP_SERVER    = 'server'
+const CONFIG_PROP_VALIDATOR = 'validate'
+const CONFIG_PROP_TEST      = 'testOnBorrow'
+
+const EVENT_CREATE_ERROR    = 'factoryCreateError'
+
+const DEFAULT_CONFIG = {
+  max               : 10
+, min               : 2
+, idleTimeoutMillis : 30000
+, log               : false
+, testOnBorrow      : false
+, logger            : console
+}
+
+
+function _addBackoff(factory) {
+  const backoff              = new FibonacciBackoff()
+  const originalCreateMethod = factory.create
+
+  factory.create = function __run() {
+
+    return originalCreateMethod().catch((e) =>
+
+      new Bluebird(function ___backoffRunResolver(resolve, reject) {
```

Comment:
Sorry to be a pest, but it looks like this `catch` should simply return `e` and chain to a promise instead of nesting the definition of a new promise inside the catch.